pulp_ostree/app/models.py

```diff
@@ -122,14 +134,39 @@ class OstreeRepository(Repository):
 
     TYPE = "ostree"
 
-    CONTENT_TYPES = [OstreeCommit, OstreeRefsHead, OstreeObject, OstreeConfig]
+    CONTENT_TYPES = [OstreeCommit, OstreeRefsHead, OstreeObject, OstreeConfig, OstreeSummary]
 
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
     def finalize_new_version(self, new_version):
-        """Handle RefsHead duplicates."""
+        """Handle repository duplicates, especially configs and summaries."""
         remove_duplicates(new_version)
+
+        if new_version.base_version:
+            previous_version = new_version.base_version
+        else:
+            if new_version.previous().number != 0:
+                previous_version = new_version.previous()
+            else:
+                previous_version = None
+
+        if previous_version:
+            new_config = new_version.added().get(pulp_type=OstreeConfig.get_pulp_type())
+            new_summary = new_version.added().get(pulp_type=OstreeSummary.get_pulp_type())
+
+            existing_files_sha256 = (
+                new_config._artifacts.get().sha256, new_summary._artifacts.get().sha256
+            )
+
+            existing_content = previous_version.added().filter(
+                _artifacts__sha256__in=existing_files_sha256
+            )
+            content_to_remove = new_version.added().filter(
+                Q(pk__in=(new_summary.pk, new_config.pk)) | ~Q(pk__in=existing_content)
```

Comment:
Wouldn't it be much easier to look if the artifacts of the new_* content had the same checksum as the present ones?

Also i am wondering what it tries to accomplish. If you intend to pretend saving existing content, it should be handled by the FindExistingContent stage.
If the rule is that only one summary / config can exist per repo version, that it should be handled by valdidate_duplicate_content when specifying proper repo_key_fields.

Reply:
Since I do not store any unique identifier for those content units, I cannot leave the whole work on FindExistingContent. Yes, the rule is to have one summary and one config per repository. But, when adding new content to a new repository version and the content **does** contain the same config and summary, the pipeline should not try to add them to the new version.

Your idea of adding `sha256` to the content types OstreeSummary and OstreeConfig is **the best** I can do to avoid this situation of checking existing content manually.

Reply:
So you actually want both.

Reply:
The summary object is optional. But, at the OSTree repository's top-level directory, there is always config (see https://fixtures.pulpproject.org/ostree/small/).